docs/rfcs/XeTile.md

```diff
@@ -140,6 +121,79 @@ A `tile_mma` variant without vector_c initialization.
+`tile_broadcast` broadcast from 1D vector to a 2D vector. 
+```mlir
+   %vector_a = XeTile.tile_broadcast %vector_b: vector<32xfloat> into vector<64x32xfloat>
+```
+
+## Internal Operations to support gradual lowering
+The 2D XeTile IR needs to be lowered in an intermediate form to support `blocking` optimization. The `blocking` optimization loads the tile in blocks and feed the block to matrix hardware. Since the load block size and matrix hardware size are not necessary same, we need to represent the data block in some form to assist the optimization. Conceptually, when a 2D tile data being loaded with a specified block size, the vector presented the 2D tile in 4D block layout. So we uses 4D dimension vector to describe the data being loaded with the block size. 
+
+`init_tile` with an `inner_block` for 2D block access of the base matrix. The `inner_blocks` attribute describes the block size for each memory load and store operation when the tile is being loaded. The block size for load may be larger than the block size for MMA operation. The output tile carries the `inner_block` attribute in its attribute set.
+
+```mlir
+  #tile_attr = #xetile.tile_attr<inner_blocks=[16,16]>
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2]:
+     memref<128x128xbf16> into tile<64x32xbf16, #tile_attr>
+```
+
+`load_tile` loads a 2D tile with an `inner_block` attribute  to 4D vector.
+```mlir
+  #tile_attr = #xetile.tile_ttr<inner_blocks=[16,16]>
+  %vector_a = XeTile.load_tile %tile_a :
+     tile<64x32xbf16, #tile_attr> into vector<4x2x16x16xb16>
+```
+`store_tile` stores a 4D vector to a 2D tile with an `inner_block`.
+```mlir
+  #tile_attr = #xetile.tile_ttr<inner_blocks=[16,16]>
+  %vector_a = XeTile.store_tile %tile_a :
+     vector<4x2x16x16xb16> into tile<64x32xbf16, #tile_attr>
+```
+`atomic_rmw_tile` performs atomic operation on 4D vectors. 
+```mlir
+#tile_attr = #xetile.tile_ttr<inner_blocks=[8,16]>
+%vector_a = atomic_rmw_tile <add> %value, %tile: vector<8x48x16xbf16>, tile<64x64xbf16, #tile_attr> to vector<8x4x8x16xbf16>
+```
+
+With the data being presented as 4D vector, all the vector based XeTile operations are required to support blocking. 
+`tile_mma` works on 4D vectors. Since dimension 1 is split into dimensions 1 and 3, the reduction of matrix multiplication is along these two dimensions.
+```mlir
+   %vector_c = XeTile.tile_mma %vector_a, %vector_b, %vector_c :
+     vector<8x8x8x16xfloat>, vector<8x4x8x8xbf16>, vector<4x8x8x16xbf16>
+	   into vector<8x8x8x16xfloat>  
+```
+`tile_reduce` follows the vector.multi-reduction semantics and can be applied to 4D vector. 
+```mlir
+   %vector_a = XeTile.tile_reduce <add> %vector_b [1, 3]: vector<8x4x8x16xfloat> into vector<8x1x8x1float>
+```
+
+`tile_broadcast` broadcast 4D vector. The input is expected to be first reshaped from 1D vector to 2D vector, and then blocked to 4D.  
+```mlir
+   %vector_a = XeTile.tile_broadcast %vector_b: vector<8x1x8x1xfloat> into vector<8x4x8x16xfloat>
```

Comment:
here it seems like the broadcast dim is listed as 1 in the input operand. So can we be consistent on the 2D case also.

1x32xf16 -> 64x32xf16